padertorch/summary/tbx_utils.py

```diff
@@ -210,42 +267,134 @@ def audio(signal, sampling_rate: int = 16000, batch_first=False,
     return signal, sampling_rate
 
 
+def figure_to_image(
+        fig: 'matplotlib.figure.Figure' = None,
+        close=True,
+) -> np.ndarray:
+    """
+    Converts a matplotlib figure to a numpy array that can be handled by
+    tensorboardX.
+
+    Uses `tensorboardX.utils.figure_to_image` with some sanity checks. It works
+    even without explicitly switching to the 'Agg' backend before using this
+    function.
+
+    Notes:
+        Use this function only if you want to convert the figure to an image
+        manually and don't want to wait for tensorboardX to convert it (e.g., if
+        you have a very large number of points that would take up too much
+        memory). Otherwise, use the `padertorch.summary.figure` function and
+        pass it under the `figures` key in the review dict.
+
+        Matplotlib is in general quite slow, so you shouldn't be computing
+        matplotlib plots in every iteration. Use the `pt.Model.create_snapshot`
+        flag to reduce the computational overhead.
+
+    Warnings:
+        Make sure that you close unused plots to reduce memory consumption by
+        either closing the plot manually or setting `close=True` (default)!
+
+    Args:
+        fig: matplotlib figure object. If `None`, it defaults to `plt.gcf()`.
+        close: If `True`, closes the figure.
+
+    Returns:
+        Numpy array with shape (color (4), height, width)
+
+    Examples:
+        >>> from matplotlib import pyplot as plt
+        >>> plt.plot([1, 3, 2, 4])  # doctest: +ELLIPSIS
+        [<matplotlib.lines.Line2D object ...>]
+        >>> image = figure_to_image()
+        >>> image.shape
+        (3, 480, 640)
+    """
+    from tensorboardX.utils import figure_to_image as tbX_figure_to_image
+    return tbX_figure_to_image(figure(fig, close=close), close=False)
+
+
+def figure(
+        fig: 'matplotlib.figure.Figure' = None,
+        close=True,
+) -> 'matplotlib.figure.Figure':
```

Comment:
I'm not sure if these type annotations make sense here. Matplotlib is not imported to keep the import overhead small, but then these annotations are not parsed correctly by PyCharm.

Reply:
Does PyCharm complain, that this is an error? Or is it just a warning?
Maybe `# noqa` can suppress pycharm's problmes.

I think the type annotation for this function is not so important. It should be obvious, what the type is.

Reply:
It's just a warning and if you use the function, it doesn't infer the return type correctly. But that's just a minor issue.